types/project.go

```diff
@@ -437,32 +464,46 @@ func (p *Project) ForServices(names []string, options ...DependencyOption) error
 			s.DependsOn = dependencies
 			enabled[name] = s
 		} else {
-			p.DisableService(s)
+			if newProject, err = newProject.DisableService(s); err != nil {
+				return nil, err
+			}
 		}
 	}
-	p.Services = enabled
-	return nil
+	newProject.Services = enabled
+	return newProject, nil
 }
 
-func (p *Project) DisableService(service ServiceConfig) {
+// DisableService removes from the project model the given service and its references in all dependencies
+// It returns a new Project instance with the changes and keep the original Project unchanged
+func (p *Project) DisableService(service ServiceConfig) (*Project, error) {
```

Comment:
minor: it's kind of weird that this one takes a `ServiceConfig` instead of a name like all the others (but that's not new)